connectors/src/connectors/notion/index.ts

```diff
@@ -553,20 +538,24 @@ export class NotionConnectorManager extends BaseConnectorManager<null> {
       }),
     ]);
 
-    const pageNodes: ContentNode[] = pages.map((page) => ({
-      provider: "notion",
-      internalId: page.notionPageId,
-      parentInternalId:
-        !page.parentId || page.parentId === "workspace" ? null : page.parentId,
-      type: "file",
-      title: page.title || "",
-      sourceUrl: page.notionUrl || null,
-      expandable: false,
-      permission: "read",
-      dustDocumentId: `notion-${page.notionPageId}`,
-      lastUpdatedAt: page.lastUpsertedTs?.getTime() || null,
-      dustTableId: null,
-    }));
+    const pageNodes: ContentNode[] = await Promise.all(
```

Comment:
Actually this promise.all is going to be super expensive no ?

Reply:
@tdraier we should definitely do this one in one query :pray: